### PR TITLE
Custom bootstrap to place nice with ipython notebook

### LINKED ISSUE #203: The notebook visualization may prevent you from restarting the kernel
I ran the mass spring damper example in PR #97 and if I "Run All" and then "Kernel>Restart" the notebook just hung in a grey screen. I had to close the browser tab and open a the notebook again to bring it back to life. There also seems to be odd behavior when "close and halt" is selected in a notebook with a viz present.

### LINKED ISSUE #209: Displaying the visualizer distorts the ipython header, etc.
calling display_ipython() causes the header to get unnecessary whitespace.

python 2.7.6 ipython 3.0.0

before
screenshot from 2015-06-08 23 25 06

after
screenshot from 2015-06-08 23 28 49

### LINKED ISSUE #227: The slider is very short
selection_092

### PR DESCRIPTION
**Pull request by sahilshekhawat**
bootstrap is already imported for ipython notebook, thus, we don't need to
Fixes #209 
Fixes #227 
Fixes #203

- [x] There are no merge conflicts.
- [x] If there is a related issue, a reference to that issue is in the
  commit message.
- [ ] Unit tests have been added for the bug. (Please reference the issue #
  in the unit test.)
- [ ] The tests pass both locally (run `nosetests`) and on Travis CI.
- [ ] The code follows PEP8 guidelines. (use a linter, e.g.
  [pylint](http://www.pylint.org), to check your code)
- [ ] The bug fix is documented in the [Release
  Notes](https://github.com/pydy/pydy#release-notes).
- [x] The code is backwards compatible. (All public methods/classes must
  follow deprecation cycles.)
- [x] All reviewer comments have been addressed.

### CODE REVIEW

File: pydy/viz/static/index_ipython.html [hunk 1]
```diff
@@ -10,7 +10,8 @@
     <link type="text/css" rel="stylesheet" href="static/css/main.css">
     <link type="text/css" rel="stylesheet" href="static/css/codemirror/blackboard.css">
     <link type="text/css" rel="stylesheet" href="static/css/slider.css">
-    <link type="text/css" rel="stylesheet" href="static/css/bootstrap.min.css">
+    <!-- commented import of bootstrap until css leak is fixed in Ipython widgets-->
+    <!--<link type="text/css" rel="stylesheet" href="static/css/bootstrap.min.css">-->
```
Comment (moorepants) [inline pydy/viz/static/index_ipython.html]: Can we just remove it if it is no longer used?

Comment (moorepants) [inline pydy/viz/static/index_ipython.html]: Why is it needed at all, even if the widgets support it?

Comment (sahilshekhawat) [inline pydy/viz/static/index_ipython.html]: We don't know how or which version they are implementing. I

Comment (sahilshekhawat) [inline pydy/viz/static/index_ipython.html]: I think to fix the leak issue they will have to fix it both ways.ie. widget's css should not be applied to the notebook's css  as well as notebook's css should not be applied to widget.

Comment (moorepants) [inline pydy/viz/static/index_ipython.html]: The point I'm trying to make is "is there any reason not to delete this line?".

**Comment by sahilshekhawat** [comment]
This needs to be review, thanks

**Comment by moorepants** [comment]
This solves the notebook header issue but it seems like the bootstrap version distributed with IPython may not have everything that we have in our distribution. Note that the glyphs are all gone and the sizing of the canvas is even different.
selection_093
 
So, either we figure out how to readjust things to that we can use the IPython boostrap or we stick with ours. Maybe using less to adjust all the css could be worthwhile, but maybe not if it takes a long time.

Also, I can imagine that this could have issues in IPython 2, too as it likely has a different version of bootstrap.

**Comment by sahilshekhawat** [comment]
Yes, trying less is probably a good idea.

**Comment by sahilshekhawat** [comment]
I have compiled bootstrap from less and it works great. Also, I found out that our html fixes best when browser's zoom is at 100%, decreasing it leads to the screenshot you showed. I think its because everything is getting changed to pixels and decreasing or increasing zoom doesn't changes it but it remains responsive.

**Comment by sahilshekhawat** [comment]
Also, After replacing our bootstrap, I am able to restart kernel with any issue.

**Comment by moorepants** [comment]
Ok, so if you complied bootstrap and you then commit it to the repo as a custom bootstrap version, this fixes both issues?

That sounds fine.

We will eventually need to have the instructions in place for compiling, but I'd be fine if you want to commit these in.

File: pydy/viz/static/css/main.css [hunk 2]
```diff
@@ -36,70 +36,74 @@ input[type=text]:focus{
...
 }
 
-ul {
+#container ul {
   list-style-type: none;
 }
+
+#container .form-control{
+	margin-bottom: 13px;
+}
```
Comment (moorepants) [inline pydy/viz/static/css/main.css]: eof

File: pydy/viz/static/index.html [hunk 3]
```diff
@@ -8,7 +8,7 @@
     <title>PyDy Visualizer</title>
 
     <!--CSS -->
-    <link type="text/css" rel="stylesheet" href="css/bootstrap.min.css">
+    <link type="text/css" rel="stylesheet" href="css/bootstrap.css">
```
Comment (moorepants) [inline pydy/viz/static/index.html]: Is there a reason we can't use the minified version?

**Comment by moorepants** [comment]
Why commit a non-minified version?

**Comment by sahilshekhawat** [comment]
Let me see if I can compile a minified version too.

**Comment by sahilshekhawat** [comment]
I can, commiting.

**Comment by moorepants** [comment]
Ok, this works for me and solves the kernel restart issue too. Nice work! Here is the screenshot with my browser zoom reset:
selection_094

**Comment by moorepants** [comment]
The play/pause/stop buttons are no longer rounded but that's fine.

**Comment by sahilshekhawat** [comment]
This is what I get on the first runtime. If you run ``scene.display_ipython`` again, it will work fine. I am trying to fix this issue.

**Comment by sahilshekhawat** [comment]
Strangely, minified version is not working properly.

**Comment by moorepants** [comment]
Well, we can commit a non-minified version. I guess it doesn't matter because minifying is only to reduce download speeds and in this case it is downloaded when you install the software. We can go with the un-mini versions.

**Comment by sahilshekhawat** [comment]
I forgot to include it in the header. It is working.

**Comment by sahilshekhawat** [comment]
Its working now, Please review it. Thanks

**Comment by moorepants** [comment]
It all looks good except the slider is still short for me.

**Comment by moorepants** [comment]
selection_095

**Comment by sahilshekhawat** [comment]
Mine is different.
pydy-ipython3

**Comment by sahilshekhawat** [comment]
Can you please check if its updated on your system or not?

**Comment by moorepants** [comment]
The slider is short in Firefox and long in Chromium. Also I can't get the canvas to load in chromium:
selection_096

**Comment by moorepants** [comment]
Chromium is definitely not working for me, here are the JS errors:
selection_097

**Comment by moorepants** [comment]
@sahilshekhawat What browser are you using?

**Comment by sahilshekhawat** [comment]
I am using chrome on linux.

**Comment by moorepants** [comment]
Any idea why chromium is totally failing for me here? It should be the same as chrome, no? I'm also on linux.

**Comment by sahilshekhawat** [comment]
None of the [three.js's examples](http://threejs.org/examples) are working in chromium on linux. So, i guess its a browser issue.

**Comment by moorepants** [comment]
So why the heck would Chrome be different than Chromium???

**Comment by sahilshekhawat** [comment]
Because chromium is open source and some of the bugs might still be there.

**Comment by moorepants** [comment]
Chrome releases are based purely off of stable Chromium releases. The only differences in the browser are some things like a pdf renderer, flash, etc.

For reference my chromium version:

Version 43.0.2357.81 Ubuntu 14.04 (64-bit)

**Comment by sahilshekhawat** [comment]
That's weird!

**Comment by moorepants** [comment]
Ok, I'll look into the Chromimum issue. But the slider is still short in Firefox. Can you test in Firefox? Once that is fixed we can merge this.

**Comment by moorepants** [comment]
What version of Chrome are you running?

**Comment by sahilshekhawat** [comment]
Yeah, I am fixing it right now.

**Comment by sahilshekhawat** [comment]
Its Version 43.0.2357.65 (64-bit)

**Comment by sahilshekhawat** [comment]
Fixed!

**Comment by moorepants** [comment]
The slider is long but this is what I'm seeing now in Firefox:
selection_098

**Comment by sahilshekhawat** [comment]
Its related to zoom in browser. You have to zoom in. The reason is that everything is converted to pixels and all the widths  based on window size.

**Comment by moorepants** [comment]
My browser zoom is at the nominal level.

**Comment by sahilshekhawat** [comment]
Then I guess that notebook has a fixed max-width and thus, a wide screen causes this issue.

**Comment by moorepants** [comment]
Ah, I have it running on a 24" monitor.

**Comment by sahilshekhawat** [comment]
I will try to find a way to fix this. Opening an issue.

**Comment by moorepants** [comment]
Ok, we can merge this as is. Thanks!

**Comment by moorepants** [comment]
I checked off the relevant boxes, leaving the rest as is.

**Pull request merged**